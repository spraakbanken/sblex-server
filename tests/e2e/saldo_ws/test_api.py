import pytest
from fastapi import status
from httpx import AsyncClient
from syrupy.matchers import path_type


@pytest.mark.asyncio
async def test_version(client: AsyncClient, snapshot_json) -> None:
    response = await client.get("/version")
    assert response.status_code == status.HTTP_200_OK
    assert response.json() == snapshot_json(
        matcher=path_type({"version": (str,), "date": (str,)})
    )


@pytest.mark.asyncio
async def test_version_json(client: AsyncClient, snapshot) -> None:
    response = await client.get("/version/json")
    assert response == snapshot


@pytest.mark.asyncio
async def test_static_saldo_css(client: AsyncClient) -> None:
    response = await client.get("/static/saldo.css")
    assert response.status_code == status.HTTP_200_OK
