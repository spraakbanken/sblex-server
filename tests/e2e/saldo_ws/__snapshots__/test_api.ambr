# serializer version: 1
# name: test_version_json
  <Response [307 Temporary Redirect]>
# ---
